src/components/settings/import-and-export/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState } from 'react'
+
+import { t } from 'i18next'
+import { Import, Share } from 'lucide-react'
+
+import { Button } from '@/components/ui/button.tsx'
+import { Label } from '@/components/ui/label'
+import { SettingsContainer } from '@/components/ui/styled.ts'
+import { Textarea } from '@/components/ui/textarea'
+import { createColor } from '@/lib/utils.ts'
+import { useStore } from '@/store'
+
+import type { Color } from '@/store/interface'
+
+export const ImportAndExportSettings = () => {
+  const cls = 'col-span-2 grid grid-cols-subgrid items-center'
+  const [style, setStyle] = useStore((store) => store.styles)
+  const [showData, setShowData] = useState('')
+  const keys = [
+    'backgroundColors',
+    'backgroundGradient',
+    'backgroundRotation',
+    'borderColors',
+    'borderGradient',
+    'borderRadius',
+    'borderRotation',
+    'borderWidth',
+    'iconColor',
+    'iconOffset',
+    'iconRotation',
+    'iconShadow',
+    'iconSize',
+    'insetShadows',
+    'padding',
+    'shadows',
+    'textColorRotation',
+    'textColors',
+    'textFont',
+    'textGradient',
+    'textItalic',
+    'textOffset',
+    'textRotation',
+    'textShadow',
+    'textSize',
+    'textValue',
+    'textWeight',
+  ]
+  return (
+    <SettingsContainer>
+      <div className={cls}>
+        <Label>{t('export data')}</Label>
+        <Button
+          size='icon'
+          onClick={() => {
+            const data = []
+            for (let i = 0; i < keys.length; i++) {
+              if (i === 0 || i === 3 || i === 17) {
+                const t: any[] = []
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-expect-error
+                style[keys[i]].forEach((item: { value: any }) => {
+                  t.push(item.value)
+                })
+                data[i] = t
+              } else if (i === 11 || i === 13 || i === 15 || i === 23) {
+                const t: any[][] = []
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-expect-error
+                style[keys[i]].forEach((item: { value: any }[]) => {
+                  t.push([item[0], item[1], item[2], item[3], item[4].value])
+                })
+                data[i] = t
+              } else {
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-expect-error
+                data[i] = style[keys[i]]
+              }
+            }
+            setShowData(JSON.stringify(data))
```

Comment:
Directly `JSON.stringify` the `style` variable should work, no need to create a new object.

Reply:
I plan to create a new object to shorten the exported data as much as possible, so I still need to use this `data` object

Reply:
If new style options are added, the export function will need changes. Using the `style` variable directly is simpler, as long as it can be serialized.

Reply:
If you add styles later, you only need to add an element after the `keys` array in the `src/components/settings/import-and-export/index.tsx` file. I don't think it's particularly troublesome.

Reply:
but this makes the code more complex, while simple serialization is enough to get the job done.

Reply:
I will update the code later and submit a PR